layers/core_checks/cc_submit_time_validation.cpp

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2024 The Khronos Group Inc.
+ * Copyright (c) 2024 Valve Corporation
+ * Copyright (c) 2024 LunarG, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cc_submit_time_validation.h"
+#include "state_tracker/queue_state.h"
+#include "sync/sync_vuid_maps.h"
+#include "generated/chassis.h"  // ValidationObject
+
+static Location GetSignaledSemaphoreLocation(const Location& submit_loc, uint32_t index) {
+    if (submit_loc.function == vvl::Func::vkQueueSubmit2) {
```

Comment:
Watch up for `vvl::Func::vkQueueSubmit2KHR`... I would just flip the logic around

```suggestion
    if (submit_loc.function != vvl::Func::vkQueueSubmit) {
```

or 

`auto field = (submit_loc.function == vvl::Func::vkQueueSubmit) ? vvl::Field::pSignalSemaphores : vvl::Field::pSignalSemaphoreInfos`

Reply:
pSignalSemaphores also work for sparse. Maybe make this more explicit.

Reply:
Updated